datumaro/datumaro/cli/contexts/project/diff.py

```diff
@@ -83,11 +83,20 @@ def save_dataset_diff(self, extractor_a, extractor_b):
         if self.output_format is Format.tensorboard:
             self.file_writer.reopen()
 
-        for i, (item_a, item_b) in enumerate(zip(extractor_a, extractor_b)):
-            if item_a.id != item_b.id or not item_a.id or not item_b.id:
-                print("Dataset items #%s '%s' '%s' do not match" % \
-                    (i + 1, item_a.id, item_b.id))
-                continue
+        ids_a = set((item.id, item.subset) for item in extractor_a)
+        ids_b = set((item.id, item.subset) for item in extractor_b)
+        ids = ids_a & ids_b
+
+        if len(ids) != len(ids_a):
+            print("Unmatched items in the first dataset: ")
```

Comment:
Why print instead of logs?

Reply:
The whole component prints some information. This should probably be put to a `json` or something similar.